server/data/data-jmap/src/main/java/org/apache/james/jmap/api/filtering/Rule.java

```diff
@@ -35,8 +35,8 @@
     public static class Id {
 
         public static Id of(String id) {
-            Preconditions.checkNotNull(id, "id should not be null");
-            Preconditions.checkArgument(StringUtils.isNotBlank(id), "id should not be empty");
+            Preconditions.checkArgument(StringUtils.isNotEmpty(id), "`id` is mandatory");
+            Preconditions.checkArgument(StringUtils.isNotBlank(id), "`id` is mandatory");
```

Comment:
By looking at the code of the StringUtils lib, it looks obvious to me that isNotBlank returns false if it is blank (so with only spaces) or also if it's null or empty. Thus isNotBlank check is enough I think?